Fix :mod:`asyncio` wait() not removing callback if exception